lib/models/fsm-transition.js

```diff
@@ -0,0 +1,660 @@
+ * associated with the current recCfg.
+ *
+ * The following is a list of the different recCfg statuses based
+ * into the different properties of recCfg itself and the associated
+ * recCfgTransition object created each time a transition from one
+ * state to another begins.
+ *
+ * - "NEW": Only the recCfg exists. None of 'staged', 'activated',
+ *   'created' or 'expired' exists. This state together with 'destroyed'
+ *   are used only to make sure all our functions work
+ *   for object instances not yet added to the backend or just removed
+ *   from it.
+ * - "CREATED": The recCfg object has been saved into persistent storage
+ *   and the value for 'created' has been added.
+ * - "STAGING": The field 'staged' hasn't been set yet, but there's
+ *   a recCfgTransition with name set to "stage", which may or not
+ *   have been started yet.
+ * - "STAGED": The aforementioned recCfg transition has finished (and
+ *   the proper 'finished' timestamp has been added to it) and the
+ *   'staged' timestamp has been set for the recovery config itself.
+ * - "UNSTAGING": A new recCfgTransition has been created with name
+ *   "unstage". It's not 'finished' yet. (When this transition finishes
+ *   the 'staged' value will be removed from the recovery config
+ *   object).
+ * - "ACTIVATING": The field 'activated' hasn't been set yet, but
+ *   there is a recCfgTranstiion with name set to "activate", which
+ *   may or not have been started yet
+ * - "ACTIVE": The aforementioned recCfg transition has finished (and
+ *   the proper 'finished' timestamp has been added to it) and the
+ *   'activated' timestamp has been set for the recovery config itself.
+ * - "DEACTIVATING": A new recCfgTransition has been created with name
+ *   "deactivate". It's not 'finished' yet. (When this transition
+ *   finishes the 'activated' value will be removed from the recovery
+ *   config object).
+ * - "EXPIRED": A given recovery configuration will enter the 'expired'
+ *   state when another one enters the 'active' state. There's no
+ *   transition for this state change. The 'expired' recCfg will reach
+ *   that state straight from 'active'. The 'expired' field will be
+ *   added to the recovery configuration record.
+ * - "REACTIVATED": There's no such 'reactivated' state, but 'created',
+ *   which can be reached by a recovery configuration in 'expired' state.
+ *   Again, there's no recCfgTransition for this state change. The values
+ *   for 'staged', 'activated' and 'expired' will be removed from the
+ *   recovery configuration record, together with any recCfgTransitions
+ *   associated with a previous life cycle.
+ * - "REMOVED": There's no such recovery configuration record any more.
+ *   The recovery configuration and all the recCfgTransitions associated
+ *   with the recovery configuration during its lifetime will be removed.
+ *
+ *   When we got a WIP transition, the only supported operation is to "CANCEL"
```

Comment:
Same it was before. What changes is that the `recCfgTransition.aborted` is set to `true` which will make the current operation to stop earlier than it would be if not cancelled.

Reply:
Could you please add a comment to the above line, to note what CANCEL does?

Reply:
Done.